okhttp/src/main/java/com/squareup/okhttp/Request.java

```diff
@@ -233,8 +233,8 @@ public Builder method(String method, RequestBody body) {
       if (body != null && !HttpMethod.permitsRequestBody(method)) {
         throw new IllegalArgumentException("method " + method + " must not have a request body.");
       }
-      if (body == null && HttpMethod.permitsRequestBody(method)) {
-        body = RequestBody.create(null, Util.EMPTY_BYTE_ARRAY);
+      if (body == null && HttpMethod.requiresRequestBody(method)) {
+        throw new IllegalArgumentException("method " + method + " must have a request body.");
```

Comment:
OH...

You're making it so that we can indeed pass empty body's, we must just explicitly create a RequestBody with an empty byte array as the content....

So I can still make `PUT` requests, we'd just want to have
```java
public Builder put() {
  return put(RequestBody.create(null, new byte[0]));
}
```